addons/mass_mailing/controllers/main.py

```diff
@@ -15,13 +15,28 @@ class MassMailController(http.Controller):
 
     def _check_mailing_email_token(self, mailing_id, document_id, email, hash_token):
         """ Return the mailing based on given credentials, sudo-ed. Raises if
-        there is an issue fetching it. """
-        if not mailing_id or not email or not document_id or not hash_token:
+        there is an issue fetching it.
+
+        Specific use case
+          * hash_token is always required for public users, no generic page if
+            available for them;
+          * hash_token is not required for generic page for logged user, aka
+            if no mailing_id is given;
+          * hash_token always requires the triplet mailing_id, email and
+            document_id, as it indicates it comes from a mailing email and
+            is used when comparing hashes;
```

Comment:
I feel like if I'm portal user I can unsubscribe the email I want, because the token is not checked in that case

Shouldn't we add a check `email == env.user.email` or something like that if we are not public ?

Reply:
Forgot what I said, in `_fetch_user_information` you return `request.env.user.email_normalized`, so all good, my bad

Reply:
Yep, it is already done like that :) I will update commit message to explain that